festim/reaction.py

```diff
@@ -113,16 +125,20 @@ def reaction_term(self, temperature):
         k = self.k_0 * exp(-self.E_k / (F.k_B * temperature))
         p = self.p_0 * exp(-self.E_p / (F.k_B * temperature))
 
-        c_A = self.reactant1.concentration
-        c_B = self.reactant2.concentration
+        reactants = self.reactant
+        product_of_reactants = reactants[0].concentration
+        for reactant in reactants[1:]:
+            product_of_reactants *= reactant.concentration
 
         if isinstance(self.product, list):
             products = self.product
         else:
             products = [self.product]
 
-        products_of_product = products[0].solution
-        for product in products[1:]:
-            products_of_product *= product.solution
-
-        return k * c_A * c_B - p * products_of_product
+        if len(products) > 0:
+            product_of_products = products[0].solution
+            for product in products[1:]:
+                product_of_products *= product.solution
+        else:
+            product_of_products = 0
```

Comment:
This line isn't tested. Please either right a test to catch it or write a #TODO to remind ourselves to test it later